crypto/rand/drbg_lib.c

```diff
@@ -33,9 +28,79 @@ static const char ossl_pers_string[] = "OpenSSL NIST SP 800-90A DRBG";
  * a much bigger deal than just re-setting an allocated resource.)
  */
 
+/*
+ * The three shared DRBGs
+ *
+ * There are three shared DRBGs (master, public and private), which are
+ * accessed concurrently by all threads.
+ *
```

Comment:
I was wondering, whether the three shared DRBGs shouldn't better be allocated on the secure heap (CRYPTO_secure_*()) instead of having them here as static variables, probably even placed in consecutive memory by the compiler.

Reply:
This makes sense.